### PR TITLE
Create FlatSelectConnected.md

### PR DESCRIPTION
**Pull request by johnnyCoveo**
### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by aiheon** [comment]
Wait for @jlevesquecoveo approval before merging :eyes:

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:10]: I didn't knew we were doing that? I thought they were only used in form like glorified radio button :see_no_evil:

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:14]: I saw that in your other pr and i'm not sure what it means? :eyes:

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:21]:
```suggestion
## Flat select vs Toggle switch
```

Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:21]:
```suggestion

## Flat select vs Toggle switch
```

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:25]:
```suggestion
- [Tabs](https://plasma.coveo.com/navigation/Tabs) Might be used instead to toggle views
```

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:26]:
```suggestion
- [Radio buttons](https://plasma.coveo.com/form/RadioButton) Can be used instead as a form input
```

File: packages/website/docs/FlatSelectConnected.md
Comment (aiheon) [inline packages/website/docs/FlatSelectConnected.md:27]:
```suggestion
- [Select - single](https://plasma.coveo.com/form/SingleSelect) Can be used instead as a form input
```

File: packages/website/docs/FlatSelectConnected.md
Comment (gdostie) [inline packages/website/docs/FlatSelectConnected.md:2]:
```suggestion
# Best Practices

The Flat select option should be used for single choice selection.
```

File: packages/website/docs/FlatSelectConnected.md
Comment (gdostie) [inline packages/website/docs/FlatSelectConnected.md:4]: It's just my opinion, but I find this part of the sentence doesn't add any value.

```suggestion
The width of each of the options should be the same, we don't want any of the options feel more important than the other.
```

File: packages/website/docs/FlatSelectConnected.md
Comment (gdostie) [inline packages/website/docs/FlatSelectConnected.md:22]: We don't have a Toggle component on plasma AFAIK, we use the Flat select for that with true/false options or a checkbox. We probably should have a toggle I guess, but we currently don't

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:10]: @aiheon we do use them as glorified radio buttons... but we shouldn't!

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:14]: @aiheon Its about the labels in the button, or am I missing the question? 😅

File: packages/website/docs/FlatSelectConnected.md
Comment (gdostie) [reply packages/website/docs/FlatSelectConnected.md:23]: I think we should until we do have the component, otherwise we advise doing something impossible

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [inline packages/website/docs/FlatSelectConnected.md:3]:
```suggestion
Use a flat select to make users select a single value among multiple options.
```

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [inline packages/website/docs/FlatSelectConnected.md:5]: @johnnyCoveo Does that mean that the width of the flat select should not change depending on the lenght of the text selected? 🤔

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [inline packages/website/docs/FlatSelectConnected.md:7]:
```suggestion
Ideally, offer a maximum of five options.
```

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [reply packages/website/docs/FlatSelectConnected.md:3]: @johnnyCoveo Also, what is the difference with a dropdown and/or single select?

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:5]: Yes, all options should have the same width!

**Review by johnnyCoveo — COMMENTED** [review]
Applied changes and remove Flat Select vs Toggle section since we don't have em yet

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [inline packages/website/docs/FlatSelectConnected.md:26]:
```suggestion
## Related Components

- [Radio buttons](https://plasma.coveo.com/form/RadioButton) - Alternative to a flat select in the context of a form
- [Single select](https://plasma.coveo.com/form/SingleSelect) - Alternative to a flat select in the context of a form
- [Tabs](https://plasma.coveo.com/navigation/Tabs) - Alternative to a toggle switch in a view switching context
```
@johnnyCoveo I feel we should give more details on what should be used when. They're probably not perfect equivalents? For instance, maybe there's a difference in the number of options, space required...?

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:9]: @jlevesquecoveo Both allows the users to make selection. Flat select exposes all the options to the users up-front and dropdown shows them in a menu when triggered. Though with the Flat select (often called Segmented Control) there is an expectation for the effect to be visible right away (think, tabs, switches).

Hope that helps!

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:11]: @jlevesquecoveo Flat select and Tabs can be very similar visually, by saying that it makes sense visually, what I mean is that, Flat select can be inside a "tab" but Tab cannot be inside a flat select... if it still doesn't make sense I think we can have a call and discuss verbally!

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:15]: @jlevesquecoveo For flat select you don't necessarily need to use verb, since its not a "button" that calls for an action. Its more of a "toggle" between things, for example in Google drive, you can switch between Grid and List view, and are labeled "Grid view" and "List view"

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:21]: @jlevesquecoveo you are right, it might be best to simply not mention toggle at all

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:26]: @jlevesquecoveo Makes sense, except line 27, we should read "Alternative to a flat select.."

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [reply packages/website/docs/FlatSelectConnected.md:11]: Gotcha, I've updated my suggestion.

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [reply packages/website/docs/FlatSelectConnected.md:15]: Ah, I see. So, any details on how to make the text scannable if it's not about verb placement in this case? 🤔 If it's under three words, I guess there's not much else you can do 😅

File: packages/website/docs/FlatSelectConnected.md
Comment (jlevesquecoveo) [reply packages/website/docs/FlatSelectConnected.md:26]: Thanks, good catch. Should we also add the checkbox, drop-down menu and toggle switch to this list, since we mention them in the article?

Also, the article didn't clarify what the difference between a single select and a flat select is. I think it should be stated here, in place of "Alternative to a flat select in the context of a form." Let me know!

File: packages/website/docs/FlatSelectConnected.md
Comment (johnnyCoveo) [reply packages/website/docs/FlatSelectConnected.md:15]: @jlevesquecoveo Simple, concise words? Hahahah 😅.. What I mean by that is that the user should be able to go through the list of options quickly and easily without too much processing effort

Comment (jlevesquecoveo) [reply packages/website/docs/FlatSelectConnected.md:15]: Okay, how about:
```suggestion
- Keep titles short, preferably **under three words**.
- Make option labels consistent and easy to scan.
```

**Review by aiheon — APPROVED** [review]

**Review by gdostie — APPROVED** [review]

**Pull request merged**
